### ISSUE TITLE
gpt-researcher 0.8.0 pip package no longer works with anthropic models

### ISSUE THREAD

**Issue by nikkag** [assafelovic/gpt-researcher#678]
Since updating the `get-researcher` package to 0.8.0, I'm getting the following exception when trying to run the researcher with Anthropic models:
```
Traceback (most recent call last):
  File "/Users/nikkag/.virtualenvs/spa/lib/python3.11/site-packages/gpt_researcher/master/actions.py", line 91, in choose_agent
    response = await create_chat_completion(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nikkag/.virtualenvs/spa/lib/python3.11/site-packages/gpt_researcher/utils/llm.py", line 88, in create_chat_completion
    provider = get_llm(llm_provider, model=model, temperature=temperature, max_tokens=max_tokens, openai_api_key=openai_api_key, **(llm_kwargs or {}))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nikkag/.virtualenvs/spa/lib/python3.11/site-packages/gpt_researcher/utils/llm.py", line 51, in get_llm
    return llm_provider(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^
TypeError: AnthropicProvider.__init__() got an unexpected keyword argument 'openai_api_key'
```
I did a quick look at it seems like `openai_api_key` was added as an argument to `create_chat_completion` and is used to instantiate the llm regardless of provider.

**Comment by leitdeux** [comment]
I've noticed that this same error happens for other non-OpenAI providers such as AzureOpenAI and Ollama as well. Adding `openai_api_key` to the parameters of `init()` of these providers seems to resolve the issue.

**Comment by assafelovic** [comment]
Hey @nikkag , just merged some fixes that might resolve this issue. Can you please pull latest and upgrade the gpt-researcher pip package? Lmk if you still encounter this issue

**Comment by nikkag** [comment]
Hey @assafelovic, thanks for addressing this! However it seems like PR #681 introduced a new issue by removing the `header` arg from the `self.retriever(sub_query)` call, without making headers an optional arg in `TavilySearch`.

```
File "/Users/nikkag/.virtualenvs/spa/lib/python3.11/site-packages/gpt_researcher/master/agent.py", line 338, in __scrape_data_by_query
    retriever = self.retriever(sub_query)
                ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: TavilySearch.__init__() missing 1 required positional argument: 'headers'
```

**Comment by nikkag** [comment]
@assafelovic Just made PR #685 to address this issue.

**Comment by gdlmx** [comment]
It seems that [this commit](https://github.com/assafelovic/gpt-researcher/commit/717e63f5d9d2a93e1b1b2f6e40f2936c8c70976f#r144410845) (5 days ago) add `openai_api_key` to every LLM provider, which causes this bug.

```diff
-     provider = get_llm(llm_provider, model=model, temperature=temperature, max_tokens=max_tokens, **llm_kwargs)
+     provider = get_llm(llm_provider, model=model, temperature=temperature, max_tokens=max_tokens, openai_api_key=openai_api_key, **(llm_kwargs or {}))
```

**Comment by ElishaKay** [comment]
Thanks for the heads up @nikkag @leitdeux @gdlmx 

Some context: The reason for adding the openai_api_key is the argument was to add support for passing the open_ai_api key directly via an API Request Header (`self.headers.get("openai_api_key")`) instead of only relying on the OS environment variable.

It seems this was resolved by [this PR](https://github.com/assafelovic/gpt-researcher/pull/681/files)

@gdlmx, are you still getting the error after cloning master or are you running via the pip package?

@assafelovic have we updated the pip_package with the PR fix above?

cc: @resvis
